### ISSUE TITLE
Tracking issue for inclusion of `derive` in lint `unused_attributes`

### ISSUE THREAD

**Issue by Havvy** [rust-lang/rust#54651]
The following code emits an unsquelchable warning.

```
#![crate_type="lib"]

#[derive()]
struct Struct;
```

```
warning: empty trait list in `derive`
 --> src/lib.rs:4:1
  |
4 | #[derive()]
  | ^^^^^^^^^^^
```

~~This warning should have a named lint `unused_derive` that is a part of the `unused` group of lints.~~

This warning should be a part of `unused_attributes`.

**Comment by petrochenkov** [comment]
And that name is `unused_attributes`.

**Comment by clintfred** [comment]
```
#[allow(unused_attributes)]
#[derive()]
```
Does not appear to squelch the warning for me...

**Comment by zackmdavis** [comment]
Implementing this as part of the existing [`UnusedAttributes` pass](https://github.com/rust-lang/rust/blob/eb50e75729bce449272ffb3bfbca2f7234f2ae13/src/librustc_lint/unused.rs#L206) would be complicated by #45216, but it might also be possible to implement where the existing "empty trait list" warning is issued (possibly as a [buffered early lint](https://github.com/rust-lang/rust/blob/eb50e75729bce449272ffb3bfbca2f7234f2ae13/src/libsyntax/early_buffered_lints.rs#L21)).

**Comment by Havvy** [comment]
I need to make `UnusedAttributes` a buffered early lint for the same thing with `cfg_attr`.

**Comment by joshtriplett** [comment]
Seems reasonable. And a macro that might end up with an empty derive list should have no problem turning off `unused_attributes` locally.

**Comment by scottmcm** [comment]
@rfcbot reviewed

I certainly agree with moving warnings to lints.
